msos/WaitChains.cs

```diff
@@ -17,68 +18,561 @@ class WaitChains : ICommand
         public int SpecificThreadId { get; set; }
 
         private CommandExecutionContext _context;
+        private BlockingObjectsStrategy _blockingObjectsStrategy;
+        private SymbolCache _symbolCache;
+        private List<UnifiedThread> _threads;
 
         public void Execute(CommandExecutionContext context)
         {
             _context = context;
+            _symbolCache = new SymbolCache();
+
+            SetStrategy();
+
             if (SpecificThreadId != 0)
             {
-                var thread = _context.Runtime.Threads.SingleOrDefault(t => t.ManagedThreadId == SpecificThreadId);
-                if (thread == null)
+                UnifiedThread thread = SpecificThread;
+
+                if (SpecificThread == null)
                 {
-                    _context.WriteError("There is no managed thread with the id '{0}'.", SpecificThreadId);
+                    _context.WriteError("There is no thread with the id '{0}'.", SpecificThreadId);
                     return;
                 }
-                DisplayChainForThread(thread);
+
+                DisplayChainForThread(thread, 0, new HashSet<int>());
+            }
+            else
+            {
+                _threads.ForEach(thread => DisplayChainForThread(thread, 0, new HashSet<int>()));
+            }
+
+            if (_context.NativeDbgEngTarget != null)
+                _context.ExitDbgEngNativeMode();
+
+        }
+
+        private void SetStrategy()
+        {
+            _threads = new List<UnifiedThread>();
+
+            if (_context.TargetType == TargetType.DumpFile)
+            {
+                if (_context.NativeDbgEngTarget == null)
+                    _context.EnterDbgEngNativeMode();
```

Comment:
How can I change it to use CreateTemporaryDbgEngTarget when DbgEngDataReader don't uses ClrMd DumpReader class?